### PR TITLE
#170947558 implements reject request feature

### PR DESCRIPTION
**Pull request by bbaime98**
#### What does this PR do?
 - add reject request functionality
#### Description of Task to be completed?
- create reject route
- create reject request function
#### How should this be manually tested?
- log in as a manager using `email: 'jules@andela.com',
        password: 'Bien@BAR789'`
- then reject a request by providing RequestId which is `0ad0ef9d-a926-4c5c-86e6-4d1e22e9ab88` and your token
- if the request is for that specific manager it will be rejected
 otherwise an error will be displayed. Reject this request `51e74db7-5510-4f50-9f15-e23710331ld5` for instance.
#### What are the relevant pivotal tracker stories?
- [#170947558](https://www.pivotaltracker.com/story/show/170947558)
#### Screenshots (if appropriate)
image

image
#### Questions:

### CODE REVIEW

**Review by izzett222 — DISMISSED** [review]

**Review by Baraka-Mugisha — DISMISSED** [review]

**Review by Ntare22 — DISMISSED** [review]

**Review by ceelogre — CHANGES_REQUESTED** [review]
Where are the unit tests?

**Comment by bbaime98** [comment]
> Where are the unit tests?

I am going to add them.

**Comment by bbaime98** [comment]
> Where are the unit tests?

Done!

File: src/tests/rejectRequest.test.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      .end((err, res) => {
+        expect(res.status).to.equal(200);
+        expect(res.body.status).to.equal(200);
+        expect(res.body.message).to.equal('Request rejected successfully');
+        done();
+      });
+  });
+  it('should find a request', () => db.Request.findOne({
+    where: { id: requestId },
+  }));
```
Comment (ceelogre) [inline src/tests/rejectRequest.test.js]: Where's assertion?

File: src/tests/rejectRequest.test.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  it('should update a request from open to rejected', () => db.Request.update({
+    status: 'rejected'
+  }, {
+    where: {
+      id: requestId,
+      managerId
+    }
```
Comment (ceelogre) [inline src/tests/rejectRequest.test.js]: Where's assertion?

File: src/tests/rejectRequest.test.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  }));
+  it('should check if rquest Id is a string', () => {
+    assert.equal(requestId, '0ad0ef9d-a926-4c5c-86e6-4d1e22e9ab88');
```
Comment (ceelogre) [inline src/tests/rejectRequest.test.js]: This is a superficial test as it's value was set on line 11. Unless it was changed by another request.

**Review by ceelogre — CHANGES_REQUESTED** [review]

**Review by izzett222 — APPROVED** [review]

**Review by Baraka-Mugisha — APPROVED** [review]

**Review by Ntare22 — APPROVED** [review]

**Review by ceelogre — APPROVED** [review]

**Pull request merged**